velox/type/Type.h

```diff
@@ -2407,3 +2424,26 @@ struct fmt::formatter<
     return formatter<string_view>::format(k->toString(), ctx);
   }
 };
+
+namespace std {
+template <>
+struct hash<facebook::velox::LongEnumParameter> {
+  size_t operator()(const ::facebook::velox::LongEnumParameter& param) const {
+    auto name = param.name;
```

Comment:
Move the implementation to .cpp file. 

It doesn't have to be std::hash. Can just be a custom hash function. You can specify it as template parameter when creating a cache.